### PR TITLE
disabled threading in numpy/scipy

### PR DESCRIPTION
**Pull request by rbooth200**
I've added a simple hack to take care of #78. It only works when using frank as a library

Is this too bad? There aren't many other options, apart from https://github.com/joblib/threadpoolctl

### CODE REVIEW

File: frank/fit.py [hunk 1]
```diff
@@ -25,6 +25,12 @@
 import sys
 import time
 import json
+
+# Force frank to run on a single thread if we are using it as a library
+if __name__ == "__main__":
+    if 'OMP_NUM_THREADS' not in os.environ:
```
Comment (mtazzari) [inline frank/fit.py]:
```suggestion

```

File: frank/fit.py [hunk 2]
```diff
@@ -25,6 +25,12 @@
...
+        os.environ['OMP_NUM_THREADS'] = '1'
```
Comment (mtazzari) [inline frank/fit.py]:
```suggestion
    os.environ.get('OMP_NUM_THREADS', '1')
```

**Comment by rbooth200** [comment]
Can you explain these changes? I'm wondering what the default behaviour should be. If not specified,I think its reasonable to set it to 1. If it is specified, maybe we should just print a warning?

**Comment by rbooth200** [comment]
I've added a warning message

**Comment by rbooth200** [comment]
Better?

**Pull request merged**